core/src/Streamly/Internal/Control/Exception.hs

```diff
@@ -86,11 +86,12 @@ verifyM predicate = verify predicate (pure ())
 -- a limited GC to free them? We can then just put gc sync barriers at points
 -- where we want to ensure that resources are freed.
 
--- | Resources with Priority1 are freed before Priority2. This is especially
--- introduced to take care of the case where we need to free channels, so that
--- all the workers of the channel are cleanup before we free the resources
--- allocated by the workers of the channel. Otherwise we might free the
--- resources and workers may be trying to use them and start misbehaving.
+-- | Resources with 'Priority1' are freed before 'Priority2'. Priority is
```

Comment:
Can you also give a small example on how this helps in freeing concurrency channels?

Reply:
A stream is being evaluated concurrently by multiple workers, each worker allocates some resource and is working with those resources. An interrupt comes and cleanup starts. Assume, we cleanup the resources of the workers first and then the worker. Now the worker is still working so it might crash or misbehave of because of the resource slipping from under it. So we need ensure that we call the cleanup of the workers before we cleanup their resources.